### ISSUE TITLE
Building off of Topic #1509 - Unable to chat in realms

### ISSUE THREAD

**Issue by jbrown1597** [killpowa/Twickt-Launcher#1540]
Hey everyone! 
Thanks for taking the time to stop by. 

I am not the author of Topic 1509, but I have read the comments and would like to bring the discussion back up, as it was closed with no resolution. 

I am experiencing the same issue (unable to chat in realms when using GDLauncher)

Using the standard launcher, chat works.
Using GDLauncher, I recieve "_Chat disabled due to missing profile public key_."

Same setup, same realm, same mods in both instances.

I've:
-restarted
-re-logged in
-resynced my time & date settings
-switched the profanity filter
-tried launching GD using 'symlink' 

- all to no avail.

Win 10
GDLauncher v1.1.30
Fabric, 1.19.3 

@nikoci @shockthetoast @GoldenAJ

Any thoughts or suggestions would be appreciated!
![Untitled - Copy](https://user-images.githubusercontent.com/50437670/216821218-09b74a56-b773-48bc-8fb7-2b6245bf67c1.jpg)

**Comment by Eskaan** [comment]
Does this happen with the Native (Microsoft) Launcher too?

**Comment by jbrown1597** [comment]
Using the standard launcher, chat does indeed still work.

**Comment by Eskaan** [comment]
Thanks.

Are you using a Fabric profile on the standard launcher too? Otherwise trying a vanilla profile from GDL might help me narrowing down the issue.

**Comment by jbrown1597** [comment]
Yes, Fabric profile on the standard launcher. I will try a vanilla instance & get back to you.

**Comment by shockthetoast** [comment]
I am out of town but just installed GDLauncher on my laptop, and only installed the Vanilla release version of 1.19.3. With that setup I am still getting the issue.

I have three other friends playing on the same realm, 2 of them use GDLauncher and get this error, one uses the standard launcher and does not.

**Comment by jbrown1597** [comment]
Apologies for the delay.

Launching a vanilla MC instance via GDLauncher still produces the same error as was mentioned before. 

Just to be certain there's no confusion between versions:
Vanilla on MC launcher: Works
Fabric on MC launcher: Works

Vanilla instance of GDLauncher: Error
Fabric instance of GDLauncher: Error

Thank you for your time & patience! Appreciate it.

**Comment by spikehidden** [comment]
I'm pretty sure the issue is regarding GDLauncher not succesfully retrieving the keys that are used to sign chat messages, as on my server in the log every message I send when using the GDLauncher is marked as unsafe. If I use the Official Launcher it is not.

**Comment by Unnon89** [comment]
Hello, i have the same problem on normal multiplayer servers. 

OS Win10
GDLauncher v1.1.30

Vanilla on MC Launcher: Works
[Fabric 1.20.1] 0.14.21: Works
[Fabric 1.20.1] 0.14.21 with Iris & Sodium on MC Launcher: Works

Vanilla instance of GDLauncher: Error
[Fabric 1.20.1] 0.14.21 instance of GDLauncher: Error
[Fabric 1.20.1] 0.14.21 with Iris & Sodium instance of GDLauncher Error

**Comment by jbrown1597** [comment]
UPDATE: Good news & bad news

Good news:
CHAT WAS FUCKING FINALLY WORKING FOR ME & A FRIEND ON REALMS TODAY!!! AHHHH

Bad news:
I had to roll my realm back a few hours & now it doesn't work again. DAMN. So close...

**Comment by jbrown1597** [comment]
UPDATE: Good news & bad news

- Good news:
CHAT WAS FUCKING FINALLY WORKING FOR ME & A FRIEND ON REALMS TODAY!!! AHHHH

- Bad news:
I had to roll my realm back a few hours & now it doesn't work again. DAMN. So close...

Prior to chat working, I had installed a mod named 'No Telemetry' 
(...which I didn't think had anything to do with any of this, but apparently, it's possible?)

- The *NEW* error message (when using "No Telemetry"):
"The server refused to deliver an unsigned message. You can enable chat signing by changing your signing mode or through a prompt screen"

When uninstalling "No Telemetry" the error message reverts back to the error from the very first post.
